ns-convert

```diff
@@ -1,184 +1,341 @@
+            self.copy_metadata(seg_group, source.metadata_raw, prefix=name)
+
+        for index in range(0, segment.item_count):
+            (data, timestamp, samples, unit) = segment.get_data(index)
+            name = '%d - %f' % (index, timestamp)
+            dset = seg_group.create_dataset(name, data=data.T)
+            dset.attrs['Timestamp'] = timestamp
+            dset.attrs['Unit'] = unit
+            dset.attrs['Index'] = index
+
+    def convert_neural(self, neural):
+        data = neural.get_data()
+        group = self._groups[neural.entity_type]
+        name = "%d - %s" % (neural.id, neural.label)
+        dset = group.create_dataset(name, data=data)
+        self.copy_metadata(dset, neural.metadata_raw)
+
+    @classmethod
+    def copy_metadata(cls, target, metadata, prefix=None):
+        for (key, value) in metadata.items():
+            if prefix is not None:
+                key = prefix + key
+            target.attrs[key] = value
+
+    @classmethod
+    def dtype_by_event(cls, event):
+        type_map = {ns.EventEntity.EVENT_TEXT  : 'a',
+                    ns.EventEntity.EVENT_CSV   : 'a',
+                    ns.EventEntity.EVENT_BYTE  : 'b',
+                    ns.EventEntity.EVENT_WORD  : 'h',
+                    ns.EventEntity.EVENT_DWORD : 'i'}
+        val_type = type_map[event.event_type]
+        if val_type == 'a':
+            val_type += str(event.max_data_length)
+        return np.dtype([('timestamp', 'd'), ('value', val_type)])
+
+
+class ConsoleIndicator(ProgressIndicator):
+    def __init__(self):
+        super(ConsoleIndicator, self).__init__()
+        self._size = 60
+        self._last_msg = ""
+
+    def progress(self,  max_value, cur_value):
+        size = self._size
+        prefix = "Converting"
+        x = int(size*cur_value/max_value)
+        msg = "%s [%s%s] %i/%i\r" % (prefix, "#"*x, "." * (size-x),
+                                     cur_value, max_value)
+        self._last_msg = msg
+        sys.stdout.write(msg)
+        sys.stdout.flush()
+
+    def cleanup(self):
+        sys.stdout.write('%s\r' % (' '*len(self._last_msg)))
+        sys.stdout.flush()
+
+
+def main():
+    global Converter,H5Converter
+    output = None
+    parser = argparse.ArgumentParser(description='Convert Neuroshare readable files to nix/hdf5 format')
+    parser.add_argument("-o","--output",type=str,help="name of the output file")
+    parser.add_argument("--hdf5",action="store_true",help="For plain conversion to hdf5 format")
+    parser.add_argument("InputFile",type=str,help="File to be converted")
+    args = parser.parse_args()
+    
+    if args.output:
+        output = args.output
+    if args.hdf5:
+        Converter = H5Converter
+    
+    filename = args.InputFile
+    ci = ConsoleIndicator()
+    converter = Converter(filename, output, progress=ci)
+    start = time.time()
+    converter.convert()
+    ci.cleanup()
+    print("converted in "+str(time.time()-start)+" seconds")

```

Comment:
Way to many digits :) "converted in 2.24568319321 seconds" maybe round to X.Y seconds? Also print it to standard error not standard out.